webapp/src/generalSlices/rsuSlice.ts

```diff
@@ -218,20 +221,52 @@ export const updateGeoMsgData = createAsyncThunk(
     const currentState = getState() as RootState
     const token = selectToken(currentState)
 
+    const requestBody = {
+      msg_type: currentState.rsu.value.geoMsgType,
+      start: currentState.rsu.value.geoMsgStart,
+      end: currentState.rsu.value.geoMsgEnd,
+      geometry: currentState.rsu.value.geoMsgCoordinates,
+    }
+
     try {
-      const geoMapData = await RsuApi.postGeoMsgData(
-        token,
-        JSON.stringify({
-          msg_type: currentState.rsu.value.geoMsgType,
-          start: currentState.rsu.value.geoMsgStart,
-          end: currentState.rsu.value.geoMsgEnd,
-          geometry: currentState.rsu.value.geoMsgCoordinates,
-        }),
-        ''
+      const geoMapData = await RsuApi.postGeoMsgData(token, JSON.stringify(requestBody), '')
```

Comment:
If this is a long running request, you can use syntax like this:
```suggestion
      const geoMapDataPromise = RsuApi.postGeoMsgData(token, JSON.stringify(requestBody), '')
      toast.promise(geoMapDataPromise, { loading: `Retrieving ${requestBody.msg_type} Data` }
      const geoMapData = await geoMapDataPromise;
```